From dfb783bf856256eb0135814194d330736884787c Mon Sep 17 00:00:00 2001
From: Cezar Craciunoiu <cezar.craciunoiu@gmail.com>
Date: Mon, 18 Jul 2022 16:02:20 +0300
Subject: [PATCH 1/8] deps/hiredis: Remove unused dependency to tcp

Signed-off-by: Cezar Craciunoiu <cezar.craciunoiu@gmail.com>
---
 deps/hiredis/sockcompat.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/deps/hiredis/sockcompat.h b/deps/hiredis/sockcompat.h
index 85810e848..f0ab1e75b 100644
--- a/deps/hiredis/sockcompat.h
+++ b/deps/hiredis/sockcompat.h
@@ -38,7 +38,7 @@
 #include <sys/select.h>
 #include <sys/un.h>
 #include <netinet/in.h>
-#include <netinet/tcp.h>
+// #include <netinet/tcp.h>
 #include <arpa/inet.h>
 #include <netdb.h>
 #include <poll.h>
-- 
2.34.1

