From d5a7436af215f6c9ff88c0afb796427be1194575 Mon Sep 17 00:00:00 2001
From: Cezar Craciunoiu <cezar.craciunoiu@gmail.com>
Date: Mon, 18 Jul 2022 16:11:56 +0300
Subject: [PATCH 4/8] src: Change header path to relative

Signed-off-by: Cezar Craciunoiu <cezar.craciunoiu@gmail.com>
---
 src/server.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/server.h b/src/server.h
index 4dcffc965..f6a0987a1 100644
--- a/src/server.h
+++ b/src/server.h
@@ -51,7 +51,7 @@
 #include <sys/socket.h>
 #include <lua.h>
 #include <signal.h>
-#include "hdr_histogram.h"
+#include "../deps/hdr_histogram/hdr_histogram.h"
 
 #ifdef HAVE_LIBSYSTEMD
 #include <systemd/sd-daemon.h>
-- 
2.34.1

